src/cosmosdb/cosmosDBConnectionStrings.ts

```diff
@@ -31,15 +29,26 @@ export class ParsedCosmosDBConnectionString extends ParsedConnectionString {
     public readonly port: string;
 
     public readonly documentEndpoint: string;
-    public readonly masterKey: string;
-
-    constructor(connectionString: string, endpoint: string, masterKey: string, databaseName: string | undefined) {
+    public readonly masterKey: string | undefined;
+
+    constructor(
+        connectionString: string,
+        endpoint: string,
+        masterKey: string | undefined,
+        databaseName: string | undefined,
+    ) {
         super(connectionString, databaseName);
         this.documentEndpoint = endpoint;
         this.masterKey = masterKey;
 
-        const parsedEndpoint = url.parse(endpoint);
-        this.hostName = nonNullProp(parsedEndpoint, 'hostname', 'hostname');
-        this.port = nonNullProp(parsedEndpoint, 'port', 'port');
+        const parsedEndpoint = new URL(endpoint);
```

Comment:
if I'm not mistaken the `new URL()` throws an error in case of wrong url besides `url.parse`

![image](https://github.com/user-attachments/assets/eb0358b8-a4fd-4769-8f87-31bbc9c8c1a0)

![image](https://github.com/user-attachments/assets/cfae5fa7-809d-4af6-9ea7-9c187a800e3a)

Reply:
@bk201- I think that's fine, the JS SDK does the same: https://github.com/Azure/azure-sdk-for-js/blob/main/sdk/cosmosdb/cosmos/src/CosmosClient.ts#L98 -> https://github.com/Azure/azure-sdk-for-js/blob/main/sdk/cosmosdb/cosmos/src/utils/checkURL.ts#L4-L6

Reply:
@bk201- See 46101db